Quaver.Shared/Screens/Gameplay/CustomAudioSampleCache.cs

```diff
@@ -26,6 +26,11 @@ public class CustomAudioSampleCache
         /// </summary>
         private List<AudioSample> Samples { get; set; } = new List<AudioSample>();
 
+        /// <summary>
+        ///     Whether the corresponding audio samples are unaffected by rate.
+        /// </summary>
+        private List<bool> UnaffectedByRate { get; set; } = new List<bool>();
```

Comment:
Instead of keeping two lists, it would be better to have one list with a class that looks as follows

```cs
public class GameplayAudioSample : IPlayableAudio, IDisposable
{
     public AudioSample Sample { get; }
     public bool UnaffectedByRate { get; }

     public GameplayAudioSample(...)
     {
     }

     public void Dispose() => Sample?.Dispose();
}
```